packages/now-cli/src/util/dns/get-dns-data.ts

```diff
@@ -25,64 +35,86 @@ export default async function getDNSData(
       .trim()
       .toUpperCase();
 
-    const name = await getRecordName(type);
-
-    if (type === 'SRV') {
-      const priority = await getNumber(`- ${type} priority: `);
-      const weight = await getNumber(`- ${type} weight: `);
-      const port = await getNumber(`- ${type} port: `);
-      const target = await getTrimmedString(`- ${type} target: `);
-      output.log(
-        `${chalk.cyan(name)} ${chalk.bold(type)} ${chalk.cyan(
-          `${priority}`
-        )} ${chalk.cyan(`${weight}`)} ${chalk.cyan(`${port}`)} ${chalk.cyan(
-          target
-        )}.`
-      );
-      return (await verifyData())
-        ? {
-            name,
-            type,
-            srv: {
-              priority,
-              weight,
-              port,
-              target
-            }
-          }
-        : null;
+    switch (type) {
+      case 'SRV':
+        return (await getData(output, type)) as SRVRecordData
+      case 'MX':
+        return (await getData(output, type)) as MXRecordData
+      case 'A':
+        return (await getData(output, type)) as ARecordData
+      case 'AAAA':
+        return (await getData(output, type)) as AAAARecordData
+      case 'ALIAS':
+        return (await getData(output, type)) as ALIASRecordData
+      case 'CAA':
+        return (await getData(output, type)) as CAARecordData
+      case 'CNAME':
+        return (await getData(output, type)) as CNAMERecordData
+      case 'TXT':
+        return (await getData(output, type)) as TXTRecordData
```

Comment:
Is it necessary to type cast `as` for all of these? The return type is still `DNSRecordData` so I think it is not necessary.

Reply:
I agree but I remove the cast TS complains about assignable types. A quick test is:
```
      case 'SRV':
        return (await getData(output, type)) as SRVRecordData
```
vs
```
      case 'SRV':
        return (await getData(output, type)) // erroring
```